perfkitbenchmarker/scripts/step_load_driver/launch_mysql_service.py

```diff
@@ -0,0 +1,258 @@
+  """
+  try:  # Parse command line flags
+    argv = FLAGS(argv)
+  except gflags.FlagsError as e:
+    logging.error('%s\nUsage: %s ARGS\n%s', e, sys.argv[0], FLAGS)
+    sys.exit(1)
+  run_uri = FLAGS.run_uri
+  if not run_uri:
+    logging.info('No run_uri given. Will run full sysbench test.')
+    run_uri = _provision_prepare_pkb()
+    logging.info('Provision and prepare completed. Run uri assigned: %s',
+                 run_uri)
+  _run(run_uri)
+  if not FLAGS.run_uri:
+    _cleanup_teardown_pkb(run_uri)
+
+
+def _provision_prepare_pkb():
+  """Run cleanup stage of PKB benchmark.
+
+  Returns:
+    run_uri: (string)
+  """
+  pkb_cmd = PKB + STAGE_FLAG + PROVISION + ',' + PREPARE
+  if FLAGS.additional_flags:
+    pkb_cmd = _append_additional_flags(pkb_cmd)
+  # PKB run with prepare,provision, wait
+  logging.info('Provision and prepare sysbench with the following command:\n%s',
+               pkb_cmd)
+  [stdout_filename, stderr_filename] = _generate_filenames(PROVISION, None)
+  _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+  return _wait_for_run(stderr_filename)
+
+
+def _run(run_uri):
+  """Run stage of PKB benchmark.
+
+  Args:
+    run_uri: (string).
+  """
+
+  run_iterations = len(FLAGS.thread_count_list)
+  logging.info(
+      'Beginning run phase. Will execute runs with %d different thread counts.',
+      run_iterations)
+  for t in FLAGS.thread_count_list:
+    pkb_cmd = (PKB + STAGE_FLAG + RUN + URI_FLAG + run_uri + THREAD_FLAG +
+               str(t) + RUN_TIME + str(FLAGS.sysbench_run_seconds) + ' &')
+    if FLAGS.additional_flags:
+      pkb_cmd = _append_additional_flags(pkb_cmd)
+    stdout_filename, stderr_filename = _generate_filenames(RUN, t)
+    logging.info('Executing PKB run with thread count: ' + str(t))
+    _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+    _wait_for_run(stderr_filename)
+    logging.info('Finished executing PKB with thread count: ' + str(t))
+
+
+def _cleanup_teardown_pkb(run_uri):
+  """Run cleanup stage of PKB benchmark.
+
+  Args:
+    run_uri: (string)
+  """
+  logging.info('Run phase complete. Starting cleanup/teardown.')
+  pkb_cmd = (PKB + STAGE_FLAG + CLEANUP + ',' + TEARDOWN + URI_FLAG + run_uri)
+  [stdout_filename, stderr_filename] = _generate_filenames(CLEANUP, None)
+  _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+
+
+def _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename):
+  """Given pkb run command, execute.
+
+  Args:
+    pkb_cmd: (str)
+    stdout_filename: (str) filename string.
+    stderr_filename: (str) filename_str
+  """
+  stdout_file = open(stdout_filename, 'w+')
+  stderr_file = open(stderr_filename, 'w+')
+  pkb_cmd_list = shlex.split(pkb_cmd)
+  logging.info('pkb command list:' + str(pkb_cmd_list))
+  subprocess.Popen(pkb_cmd_list, stdout=stdout_file, stderr=stderr_file)
+
+
+def _wait_for_run(filename):
+  """Given a filename, wait for it to be populated, return run_uri.
```

Comment:
Realized the subprocess call I'm making has a clean way of way of waiting for process to terminate. Testing it now. Now the wait_for_run will only grab the run_uri for provision/prepare once the process has completed.